### PR TITLE
Added Jenkinsfile for continuous integration

### PR DESCRIPTION
**Pull request by ghost**
Added a Jenkins file at root
Currently runs and reports:
-Valgrind
-Nosetests, gcovr
-cppcheck, scanbuild
-SLOCcount, pep8, pylynt

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/groovy
+node {
+    env.NOSETESTS_ARGS=" --where=tests --match=\".*\\.((py)|c)\" --with-xunit --verbose --process-restartworker --process-timeout=60 --xunit-file=$WORKSPACE/tests/report/nosetests.xml"
+    env.NOSETESTS_ARGS="$NOSETESTS_ARGS --with-coverage --cover-xml --cover-xml-file=$WORKSPACE/tests/report/py_coverage.xml --cover-html --cover-html-dir=$WORKSPACE/tests/report/py_html"
+    env.NOSETESTS_ARGS="$NOSETESTS_ARGS --cover-inclusive  --cover-package=bxi"
+    env.VALGRIND_ARGS="--fair-sched=yes --child-silent-after-fork=yes --tool=memcheck --xml=yes --xml-file=$WORKSPACE/tests/report/valgrind/valgrind-%p.xml --leak-check=full --show-leak-kinds=all"
+    env.GCOVR="gcovr -v -r $WORKSPACE -e '.*usr/include' -e '.*local.*' -e '.*_.*' -x -o $WORKSPACE/tests/report/c_coverage.xml"
+    env.PYTHONCONF="PYTHON=/usr/bin/python3"
```
Comment (jnquintin) [inline Jenkinsfile]: Could we take this from the environment?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    sed -i \"s#/#.#g\" $WORKSPACE/modules.name
+    PYTHONPATH=$WORKSPACE/packaged/lib pylint -f parseable --rcfile=$WORKSPACE/misc/shared/pylint.rc \$(cat modules.name) > pylint.txt || echo 'pylint complete'
+    '''
+    }
+
+    stage('Test') {
+	echo 'Testing..'
+	sh '''
+	cd install/
+	. $WORKSPACE/dependencies.sh
```
Comment (jnquintin) [inline Jenkinsfile]: This file is missing in the pull request no?

Comment (jnquintin) [reply Jenkinsfile]: Could we have the folder as a parameter instead of moving in the folder?

File: Jenkinsfile (hunk 1 above)
Comment (ghost) [reply Jenkinsfile]: I added a condition to take this from env if it's already set

**Pull request closed without merging**